docs/content/guides/dagster/local_to_production_tutorial.mdx

```diff
@@ -0,0 +1,395 @@
+                    "database": "SANDBOX",
+                    "schema": "JAMIE"
+            })
+        },
+        "production": {
+            "hn_client": hn_api_client,
+            "io_manager": snowflake_io_manager.configured({
+                    "account": "abc1234.us-east-1",
+                    "user": "dev@company.com",
+                    "password": "company_super_secret_password",
+                    "database": "PRODUCTION",
+                    "schema": "HACKER_NEWS"
+            })
+        }
+    }
+    deployment_name = os.environ.get("DAGSTER_DEPLOYMENT", "local")
+
+    return [
+        with_resources([items, comments, stories], resource_defs=resource_defs[deployment_name])
+    ]
+```
+
+Now we just set the environment variable `DAGSTER_DEPLOYMENT=production` in our deployment and the correct resources will be applied to the assets.
+
+<Note>
+
+If you are using \[Dagster Cloud]\(link to dagster cloud page explaining this?), this environment variable will be set for you. Additionally, if you are using \[Branch Deployments]\(link to branch deployments page explaining this?) `DAGSTER_DEPLOYMENT` will be automatically set to `VAR`.
+
+</Note>
+
+We still have some problems with this setup:
+
+1. Each developer still needs to remember to change `user` and `password` to their credentials and `schema` to their name when they develop locally.
+2. Passwords are being stored in code.
+3. If we want to change the database where we store development or production data, we need to make a code change.
+
+We can easily solve all of these problems because the snowflake IO manager accepts configuration from environment variables using the <PyObject object="StringSource" /> configuration type. This allows us to store configuration values as environment variables, and point dagster to those environment variables in the run configuration:
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v5.py startafter=start endbefore=end
+# repository.py
+
+@repository
+def repo():
+    resource_defs = {
+        "hn_client": hn_api_client,
+        "io_manager": snowflake_io_manager.configured(
+            {
+                "account": {"env": "SNOWFLAKE_ACCOUNT"},
+                "user": {"env": "SNOWFLAKE_USER"},
+                "password": {"env": "SNOWFLAKE_PASSWORD"},
+                "database": {"env": "SNOWFLAKE_DATABASE"},
+                "schema": {"env": "SNOWFLAKE_SCHEMA"},
+            }
+        ),
+    }
+
+    return [*with_resources([items, comments, stories], resource_defs=resource_defs)]
+```
+
+<Note>
+
+Since all of the snowflake IO manager configuration is being set through environment variables, we no longer need to select the resource configuration based on deployment! If we later add more configuration that varies between local and production, we can go back to the previous pattern of selecting config based on the deployment name.
```

Comment:
```suggestion
Since all of the Snowflake I/O manager configuration is being set through environment variables, we no longer need to select the resource configuration based on deployment! If we decide to add more configuration that varies between local and production, we can go back to the previous pattern of selecting config based on the deployment name.
```